### PR TITLE
fix(android, ios): eliminates OSK layout flashing from predictive text banner display

### LINKED ISSUE #3294: bug(web): Keyman loading time and banner splashing in iOS and Android
**Describe the bug**
Keyman loading is slow and banner splashes. This can be avoided.

**To Reproduce**
Switch the keyboard in iOS and android from system to Keyman.

**Expected behavior**
1. quick loading
2. No splashing of banners

**Screenshots**
Screen capture video is attached.
[ScreenCapture.zip](https://github.com/keymanapp/keyman/files/4842856/ScreenCapture.zip)

---
  
**iPhone:**
 - Device: [e.g. iPhone 11 pro max]
 - OS: [e.g. iOS 13.5.1]
 - Keyman version [e.g. 14.0.101]
 - Target application: [e.g. Notes, Mail]

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #3294.

- Keyboards with associated dictionaries will now load more smoothly.

There will still be a small amount of layout calibration / FOUC-ish behavior on the initial load of each keyboard during the app / app extension's lifetime.  The egregious shift caused by the banner appearing late has been fixed.  (The banner's suggestions, on the other hand, will arrive at the original, 'late', time.)

Further keyboard swapping will be 100% smooth for as long as the system keeps the app/keyboard/extension alive.

### User Testing
@MakaraSok (or others, if so inclined)

Could use a user test on Android; I've already tested with iOS.
- While I _doubt_ it'll matter, a bit of testing with the keyboard-based keyboard-swapping shortcuts may not be a bad idea, just in case.
- Be sure to test that keyboard swapping works correctly both in-app and when used as the system keyboard.
- Does the banner display properly every time you expect it?
    - Note:  initial suggestions will likely appear 'late' after a swap.  _That_ can't be helped.
- Does the banner display at the same time as the main keyboard, or sometime later?  (Should be simultaneous)

### CODE REVIEW

File: common/core/web/input-processor/src/text/prediction/languageProcessor.ts [hunk 1]
```diff
@@ -137,17 +137,29 @@ namespace com.keyman.text.prediction {
...
-        lp.currentModel = model;
         lp.configuration = config;
-
-        try {
-          lp.emit('statechange', 'active');
-        } catch (err) {
-          // Does this provide enough logging information?
-          console.error("Could not load model '" + model.id + "': " + (err as Error).message);
+        lp.emit('statechange', 'configured');
+      }).catch(function(error) {
```
Comment (jahorton) [inline common/core/web/input-processor/src/text/prediction/languageProcessor.ts:149]: The big change + fix.

In particular, note `lp.emit('statechange', 'active');`, which emits an event that triggers display of the OSK's suggestion banner.

Originally, this event was only emitted once a model fully initialized... which requires deferment through a `Promise`.  That deferment is the cause of the 'layout splashing' effect that's been seen in our Android & iOS keyboards for a while now, as the banner simply _will not_ be available for display at the same time as the main body of the OSK.

By instead pre-emptively declaring a model `active` (which needs the associated `currentModel` assignment), we can synchronously set the banner.  Doing so naively does fail to produce initial suggestions, which is what the new `'configured'` state-change event flag is for.

**Comment by MakaraSok** [comment]
Tested on Android 10 of an emulator and physical device.

The banner is indeed shown at the same time as the keyboard layout, however the characters shown on the layout seem to take a little longer to load (flicked once if not twice), the same goes for the suggested words in the banner.

**Review by mcdurdin — APPROVED** [review]
Changes LGTM. I have not yet tested, but will keep trying it on my device when the changes land in beta.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.233-beta](https://keyman.com/downloads/releases/beta/14.0.233)

**Comment by jahorton** [comment]
> Changes LGTM. I have not yet tested, but will keep trying it on my device when the changes land in beta.

@mcdurdin - just a reminder to give it a try when you have the time.

**Comment by mcdurdin** [comment]
This looks much, much better. Good work. Big step forward.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.15-alpha](https://keyman.com/downloads/releases/alpha/15.0.15)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.16-alpha](https://keyman.com/downloads/releases/alpha/15.0.16)

**Pull request merged**